testnet/stacks-node/src/tests/signer/commands/block_wait.rs

```diff
@@ -123,3 +129,139 @@ impl Command<SignerTestState, SignerTestContext> for WaitForTenureChangeBlockFro
         ))
     }
 }
+
+/// ---------------------------------------------------------
+/// ---------------------------------------------------------
+/// ---------------------------------------------------------
+/// ---------------------------------------------------------
+/// ---------------------------------------------------------
+/// Draft version of the command
+
+pub struct WaitForRejectedBlockProposal {
+    miners: Arc<Mutex<MultipleMinerTest>>,
+    timeout_secs: u64,
+    num_signers: usize,
+}
+
+impl WaitForRejectedBlockProposal {
+    pub fn new(
+        miners: Arc<Mutex<MultipleMinerTest>>,
+        timeout_secs: u64,
+        num_signers: usize,
+    ) -> Self {
+        Self {
+            miners,
+            timeout_secs,
+            num_signers,
+        }
+    }
+}
+
+impl Command<SignerTestState, SignerTestContext> for WaitForRejectedBlockProposal {
+    fn check(&self, state: &SignerTestState) -> bool {
+        // We need multiple preconditions to be met:
+        // 1. We've booted to Nakamoto
+        // 2. The primary miner should have skipped commit operations (leading to potential rejections)
+        // 3. We're not in a mining stalled state (we want active proposals)
+        let result = state.is_booted_to_nakamoto
+            && state.is_primary_miner_skip_commit_op
+            && !state.mining_stalled;
+
+        info!(
+            "Checking: Wait for rejected block proposal with timeout {} seconds. Results: nakamoto={}, primary_skip={}, mining_active={}. Overall={}",
+            self.timeout_secs,
+            state.is_booted_to_nakamoto,
+            state.is_primary_miner_skip_commit_op,
+            !state.mining_stalled,
+            result
+        );
+
+        result
+    }
+
+    fn apply(&self, state: &mut SignerTestState) {
+        info!(
+            "Applying: Wait for rejected block proposal with timeout {} seconds",
+            self.timeout_secs
+        );
+
+        // Get the configs and miner keys
+        let miners_guard = self.miners.lock().unwrap();
+        let (conf_1, _) = miners_guard.get_node_configs();
+        let (miner_pk_1, _) = miners_guard.get_miner_public_keys();
+        drop(miners_guard); // Release the lock
```

Comment:
This is a nit. I would go for `miners` instead of `miners_guard`, for consistency with the rest of the places:
```suggestion
        // Get the configs and miner keys
        let miners = self.miners.lock().unwrap();
        let (conf_1, _) = miners.get_node_configs();
        let (miner_pk_1, _) = miners.get_miner_public_keys();
        drop(miners); // Release the lock
```